doc/sonic-build-system/rfs-split-build-improvement.md

```diff
@@ -0,0 +1,83 @@
+# RFS split build improvements HLD
+
+# Table of Contents
+
+- [Revision History](#revision-history)
+- [Abbreviations](#abbreviations)
+- [Overview](#overview)
+    - [10,000 foot view of SONiC installer build](#10,000-foot-view-of-SONiC-installer-build)
+- [RFS Split](#rfs-split)
+    - [Implementation details](#implementation-details)
+
+# Revision History
+| Rev |    Date  |     Author      | Change Description                                           |
+|:---:|:--------:|:---------------:|:------------------------------------------------------------:|
+| 0.1 |          | Yakiv Huryk     | Initial version                                              |
+
+### Abbreviations
+
+| **Term** | **Meaning**                               |
+| -------- | ----------------------------------------- |
+| RFS      | Root Filesystem                           |
+
+# Overview
+
+The goal of this change is to optimize the SONiC installer build by splitting the final build step into two stages. This will allow running the first stage in parallel, improving build time.
+
+## 10,000 foot view of SONiC installer build
+The very last step of the SONiC build process is to build the $(SONIC_INSTALLERS) target, which is usually defined in platform one-image.mk file:
+```makefile
+SONIC_ONE_IMAGE = sonic-mellanox.bin
+$(SONIC_ONE_IMAGE)_MACHINE = Mellanox
+...
+SONIC_INSTALLERS += $(SONIC_ONE_IMAGE)
+
+```
+
+There is a designated target in slave.mk to build it:
+```makefile
+# targets for building installers with base image
+$(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% :
+```
+The rule builds/prepares a Debian image and installs all necessary artifacts (debian packages, docker images, files, etc.). Since this rule depends on all the other build rules, it cannot be parallelized. This means that if we change some dependencies, in order to build the image, the build system needs to do all the steps in this rule from scratch.
```

Comment:
```
This means that if we change some dependencies, in order to build the image, the build system needs to do all the steps in this rule from scratch.
```  

Even after the new changes today, if we start a new build the first stage rootfs will be built again. So this statement doesn't hold true. I think it's okay to remove this statement. The important advantage is Parallel Execution of First stage, which is covered

Reply:
Good point, removed